ipfs-8-to-9/migration/migration.go

```diff
@@ -0,0 +1,142 @@
+package mg8
+
+import (
+	base32 "encoding/base32"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	migrate "github.com/ipfs/fs-repo-migrations/go-migrate"
+	mfsr "github.com/ipfs/fs-repo-migrations/mfsr"
+	log "github.com/ipfs/fs-repo-migrations/stump"
+)
+
+type Migration struct{}
+
+func (m Migration) Versions() string {
+	return "8-to-9"
+}
+
+func (m Migration) Reversible() bool {
+	return true
+}
+
+const keyFilenamePrefix = "key_"
+
+const keystoreRoot = "keystore"
+
+func isEncoded(name string) bool {
```

Comment:
Given that std base32 includes all letters, isn't this migration a no-op for any key named with a single lower-case word? (`main`, `ipns`, etc.).?

If you want to be robust, we should copy keys from an old directory to a new directory, then replace the old directory. However, even that isn't perfect as we might fail between replacing the old directory and updating the repo version.

Really, we need some way to record that a migration is in-progress. But we don't currently have a way to do that.

Reply:
Decode (and isEncoded()) will fail if the filename do not start with `key_`. Not 100% bulletproof, but exceptions will be logged.

I think this is an ok assumption to make, mostly because repairing the issue if it happens is easy for the user.

Reply:
Ah, now I see. Nevermind.